mobile-app/lib/ui/views/learn/block/templates/dialogue/dialogue_view.dart

```diff
@@ -0,0 +1,181 @@
+            child: ListView.builder(
+              physics: const ClampingScrollPhysics(),
+              shrinkWrap: true,
+              itemCount: structure.length,
+              itemBuilder: (context, dialogueBlock) {
+                return Column(
+                  children: [
+                    Row(
+                      children: [
+                        Expanded(
+                          child: InkWell(
+                            onTap: () => model.routeToChallengeView(
+                              block,
+                              dialogueHeaders[dialogueBlock].id,
+                            ),
+                            child: Container(
+                              decoration: BoxDecoration(
+                                borderRadius: BorderRadius.circular(5),
+                                color: model.completedChallenge(
+                                  dialogueHeaders[dialogueBlock].id,
+                                )
+                                    ? const Color.fromRGBO(
+                                        0x00, 0x2e, 0xad, 0.3)
+                                    : const Color.fromRGBO(0x2a, 0x2a, 0x40, 1),
+                                border: Border.all(
+                                  color: model.completedChallenge(
+                                          dialogueHeaders[dialogueBlock].id)
+                                      ? const Color.fromRGBO(
+                                          0xbc, 0xe8, 0xf1, 1)
+                                      : const Color.fromRGBO(
+                                          0x3b, 0x3b, 0x4f, 1),
+                                  width: 1,
+                                ),
+                              ),
+                              child: Padding(
+                                padding: const EdgeInsets.all(8.0),
+                                child: Text(
+                                  dialogueHeaders[dialogueBlock].title,
+                                  style: const TextStyle(
+                                    fontSize: 18,
+                                    fontWeight: FontWeight.bold,
+                                  ),
+                                ),
+                              ),
+                            ),
+                          ),
+                        ),
+                      ],
+                    ),
+                    SizedBox(
+                      width: MediaQuery.of(context).size.width - 34,
+                      height: 200,
+                      child: Padding(
+                        padding: const EdgeInsets.all(8.0),
+                        child: GridView.builder(
+                          itemCount: structure[dialogueBlock].length,
+                          gridDelegate:
+                              const SliverGridDelegateWithFixedCrossAxisCount(
+                            crossAxisCount: 6,
+                            mainAxisExtent: 60,
```

Comment:
Not needed as this just sets the height of the grid cell and without it it'll automatically be a square as the default aspect ratio is 1
```suggestion
```